GameOfRunes/GameOfRunes/Types/PowerUpType.swift

```diff
@@ -8,8 +8,70 @@
 
 import SpriteKit
 
-enum PowerUpType: String {
+enum PowerUpType: String, CaseIterable {
     case hellfire
     case icePrison
     case darkVortex
+    
+    private static var powerUpCastsTextures = [PowerUpType: [SKTexture]]()
+    private static var powerUpsTextures = [PowerUpType: [SKTexture]]()
+    
+    // have to preload at the start of the gameplay scene to prevent delay in rendering
+    static func loadPowerUpCastsTextures() {
+        guard powerUpCastsTextures.isEmpty else {
+            return
+        }
+        allCases.forEach { powerUpType in
+            let powerUpCastAtlas = SKTextureAtlas(named: "\(powerUpType)Cast")
+            let powerUpCastTextures = (0...19).map {
+                powerUpCastAtlas.textureNamed(.init(format: "tile%03d", $0))
+            }
+            powerUpCastsTextures[powerUpType] = powerUpCastTextures
+        }
+        
+        loadPowerUpTextures()
+    }
+    
+    private static func loadPowerUpTextures() {
+        guard powerUpsTextures.isEmpty else {
+            return
+        }
+        let hellfireAtlas = SKTextureAtlas(named: hellfire.rawValue)
+        powerUpsTextures[hellfire] = (690000...690019).map { hellfireAtlas.textureNamed("\($0)") }
+        
+        let darkVortexAtlas = SKTextureAtlas(named: darkVortex.rawValue)
+        powerUpsTextures[darkVortex] = (670000...670019).map { darkVortexAtlas.textureNamed("\($0)") }
+    }
+    
+    func runAnimation(at position: CGPoint, with size: CGSize, on node: SKNode) {
+        let animationNode = SKSpriteNode(texture: nil, color: .clear, size: size)
+        animationNode.position = position
+        var powerUpCastAnimation: SKAction?
+        var powerUpAnimation: SKAction?
+        
+        if let powerUpCastTextures = Self.powerUpCastsTextures[self] {
```

Comment:
do we need the extra `Self`?

Reply:
Yepp, the static variable is accessed from a non-static context